packages/components/src/styles/components/app-frame.scss

```diff
@@ -16,32 +18,39 @@
   grid-template-rows: auto 1fr auto;
   grid-template-columns: auto 1fr;
   min-height: 100vh;
-
-  // prevent interaction with AppHeader when SideNav is open/expanded
-  &:has(.hds-side-nav--is-not-minimized) .hds-app-header--is-mobile {
-    pointer-events: none;
-  }
 }
 
-
 // FRAME'S CONTAINERS
 
 .hds-app-frame__header {
-  z-index: 7;
+  z-index: 21; // needs to stay above the main content & sidebar
   grid-area: header;
 
-  // NOTE: May not need to qualify this with .hds-app-header as it may not matter if AppHeader is not present
-  &:has(.hds-app-header) {
+  // Make header position sticky/fixed if the viewport height is greater than 480px
+  @media (height >= 480px) {
     position: sticky;
     top: 0;
     right: 0;
     left: 0;
+
+    // Modify sidenav layout when used together with fixed app-header
+    &:has(.hds-app-header) {
+      + .hds-app-frame__sidebar {
```

Comment:
[thought/non-blocking] for me having this here is confusing: it took me a while to realize this rule applies to the sidebar not the header (it's in the "header" block).

an option would be to move it down inside the `.hds-app-frame__sidebar`, something like this (rough code):

```

.hds-app-frame__sidebar {
  // rules here

  @media (height >= 480px) {
    .hds-app-frame__header:has(.hds-app-header) + & {
      // top/height/min-height here
    }
  }
}
```

Another option to explore is to see if we could use instead the `hasHeader`/`hasSidebar` booleans to determine when both are present, and add a class accordingly (in which case, the `classNames` getter can be left in the backing class).

Reply:
Added a conditional classname (.hds-app-frame--has-header-with-sidebar) to the AppFrame and used it to enable positioning related styles.